nixos/modules/system/activation/bootspec.nix

```diff
@@ -79,9 +75,18 @@ let
 in
 {
   options.boot.bootspec = {
-    enable = lib.mkEnableOption (lib.mdDoc "Enable generation of RFC-0125 bootspec in $system/bootspec, e.g. /run/current-system/bootspec");
+    enable = lib.mkEnableOption (lib.mdDoc "the generation of RFC-0125 bootspec in $system/boot.json, e.g. /run/current-system/boot.json")
+    // { default = true; };
```

Comment:
Shouldn't this be enabled by each boot loader that needs this?

Reply:
Well, the idea is that the bootspec is always available so tools can rely on it any time after this PR is merged.

Then, tools don't need to synthesize a bootspec for generations older than the first one that enabled tool A, but instead only for generations older than the first one that contains this PR.

Reply:
Ok, if it's expected to be always available then the option can be removed or made internal.

Reply:
> Shouldn't this be enabled by each boot loader that needs this file?

If we do this, each time someone discovers they want to enable bootspec, they will have to discard all their previous generations because the information cannot be generated.

Reply:
> Ok, if it's expected to be always available then the option can be removed or made internal.

Indeed.

Reply:
I would rather opt for internal.

Reply:
I mean, there may be valid reasons people may want to disable it. I don't know what they could be, but they get to keep both pieces if something breaks because of it.

Internal is more sensible, but removes it from the manual (and maybe even option documentation, so the changelog link will not work anymore).